### PR TITLE
Update documentations for the latest ngen

### PR DESCRIPTION
**Pull request by stcui007**
This PR updates several documentations so that the build of module libraries, test modules, and ngen build options and environment reflect the most recent refactoring in the CMake build structure of ngen. In addition, it updates the routing_config.yaml so that it works with the current t-route. The previous version no longer works. It also add a few tips for building the nwm-routing module.

## Additions

-

## Removals

-

## Changes

data/gauge_01073000/routing_config.yaml
doc/BMI_MODELS.md
doc/DEPENDENCIES.md
doc/DISTRIBUTED_PROCESSING.md
doc/MPI_REMOTE_NEXUS.md
doc/REALIZATION_CONFIGURATION.md
doc/PYTHON_ROUTING.md
test/README.md
./README.md
./INSTALL.md

## Testing

1. Browser

## Notes

Work in progress

## Todos

-

## Checklist

- [x ] PR has an informative and human-readable title
- [x ] Changes are limited to a single goal (no scope creep)
- [x ] Code can be automatically merged (no conflicts)
- [x ] Code follows project standards (link if applicable)
- [x ] Passes all existing automated tests
- [x ] Any _change_ in functionality is tested
- [x ] New functions are documented (with a description, list of inputs, and expected output)
- [x ] Placeholder code is flagged / future todos are captured in comments
- [x ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)
- [x ] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

## Testing checklist (automated report can be put here)

1.

### Target Environment support

- [x ] Linux

### CODE REVIEW

File: doc/REALIZATION_CONFIGURATION.md [hunk 1]
```diff
@@ -32,7 +32,7 @@ The `global` key-value object must contain the following two object keys:
   * Note: future versions could support breaking up `params` into additional key-value subobjects for `options` and `initial_conditions`
   * `params` must be a list that holds key-value pairs
 * `forcing`
-  * key-value object with keys for `file_pattern` and `path` that define the default CSV file pattern and path for the input forcings relative to the executable directory
+  * key-value object with keys for `file_pattern` and `path` that define the default CSV file pattern and path for the input forcings relative to the executable directory. More recently, `ngen` developed capability to handle forcing data in different format. Thus, a `provider` value parameter can be used to explicitly define the format of the forcing data, such as NetCDF format, in the form "provider": "NetCDF".
```
Comment (program--) [inline doc/REALIZATION_CONFIGURATION.md:35]:
```suggestion
  * key-value object with keys for `file_pattern` and `path` that define the default CSV file pattern and path for the input forcings relative to the executable directory. More recently, `ngen` developed the capability to handle forcing data in different formats. Thus, a `provider` value parameter can be used to explicitly define the format of the forcing data, such as NetCDF format, in the form "provider": "NetCDF".
```

Comment (stcui007) [reply doc/REALIZATION_CONFIGURATION.md:35]: Thanks. I made the corrections as you suggested. To be uploaded.

**Review by program-- — APPROVED** [review]

File: doc/DEPENDENCIES.md [hunk 2]
```diff
@@ -6,15 +6,16 @@
 | ------------- |:------------- | :-----:| :-------: |
 | [Google Test](#google-test) | submodule  | `release-1.10.0` | |
 | [C/C++ Compiler](#c-and-c-compiler) | external | see below |  |
-| [CMake](#cmake) | external | \>= `3.14` | |
+| [CMake](#cmake) | external | \>= `3.15` | |
```
Comment (program--) [inline doc/DEPENDENCIES.md:9]: Didn't catch this before, but the minimum version required for CMake is 3.17, see the root `CMakeLists.txt`.

**Comment by stcui007** [comment]
Oh, thank you. I got that from git ngen-build yaml. I am changing it now.

**Review by program-- — APPROVED** [review]

**Pull request merged**